### ISSUE TITLE
Broken special remote lets whereis fail completely

### ISSUE THREAD

**Issue by bpoldrack** [datalad/datalad#4204]
Scenario: We have a repo with a file locally available and (according to current knowledge) available from that special remote. Now we break the special remote, so that it responds to PREPARE by PREPAIRE-FAILURE. Effect: `git-annex-fsck` as well as `git-annex-whereis` fail. While the former is fine, the latter isn't, since due its failure we don't get *any* response from `whereis` anymore. Not even the local availability:

```
❱ git annex --debug whereis one.txt --json                                                                                                                                               1 !
[2020-02-26 12:26:59.771530846] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","show-ref","git-annex"]
[2020-02-26 12:26:59.780514055] process done ExitSuccess
[2020-02-26 12:26:59.780884245] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","show-ref","--hash","refs/heads/git-annex"]
[2020-02-26 12:26:59.788015914] process done ExitSuccess
[2020-02-26 12:26:59.788589443] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","log","refs/heads/git-annex..d4d0c919ff3a0bf7e604b1c94ea98feefeb1b684","--pretty=%H","-n1"]
[2020-02-26 12:26:59.795904432] process done ExitSuccess
[2020-02-26 12:26:59.796823695] chat: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","cat-file","--batch"]
[2020-02-26 12:26:59.812617559] chat: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","cat-file","--batch-check=%(objectname) %(objecttype) %(objectsize)"]
[2020-02-26 12:26:59.819050683] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","symbolic-ref","-q","HEAD"]
[2020-02-26 12:26:59.825218144] process done ExitSuccess
[2020-02-26 12:26:59.825441474] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","show-ref","refs/heads/master"]
[2020-02-26 12:26:59.83136621] process done ExitSuccess
[2020-02-26 12:26:59.831736442] read: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","ls-files","--cached","-z","--","one.txt"]
[2020-02-26 12:26:59.838333917] chat: /home/ben/venvs/datalad-dev/bin/git-annex-remote-ria []
[2020-02-26 12:26:59.972062944] git-annex-remote-ria[1] --> VERSION 1
[2020-02-26 12:26:59.972170997] git-annex-remote-ria[1] <-- EXTENSIONS INFO
[2020-02-26 12:26:59.972385856] git-annex-remote-ria[1] --> EXTENSIONS
[2020-02-26 12:26:59.972435141] git-annex-remote-ria[1] <-- PREPARE
[2020-02-26 12:26:59.972654264] git-annex-remote-ria[1] --> GETGITDIR
[2020-02-26 12:26:59.972700314] git-annex-remote-ria[1] <-- VALUE .git
[2020-02-26 12:26:59.972801147] git-annex-remote-ria[1] --> GETUUID
[2020-02-26 12:26:59.972851531] git-annex-remote-ria[1] <-- VALUE e1ee313a-1f2b-4ab9-b10d-c35407c85072
[2020-02-26 12:26:59.972967572] git-annex-remote-ria[1] --> GETCONFIG name
[2020-02-26 12:26:59.973021609] git-annex-remote-ria[1] <-- VALUE archive
[2020-02-26 12:26:59.973087917] git-annex-remote-ria[1] --> GETCONFIG url
[2020-02-26 12:26:59.973125401] git-annex-remote-ria[1] <-- VALUE ria+file:///tmp/datalad_temp_test_backup_archive4p_w2fdc
[2020-02-26 12:27:00.001862614] git-annex-remote-ria[1] --> GETCONFIG archive-id
[2020-02-26 12:27:00.002125072] git-annex-remote-ria[1] <-- VALUE c516c60c-588a-11ea-9596-e11c00cb83c9
[2020-02-26 12:27:00.002445627] git-annex-remote-ria[1] --> GETCONFIG force-write
[2020-02-26 12:27:00.002551097] git-annex-remote-ria[1] <-- VALUE 
[2020-02-26 12:27:00.003791357] git-annex-remote-ria[1] --> PREPARE-FAILURE 
git-annex: 
{"command":"whereis","success":false,"key":"MD5E-s8--7e55db001d319a94b0b713529a756623.txt","file":"one.txt"}
[2020-02-26 12:27:00.037076997] process done ExitSuccess
[2020-02-26 12:27:00.03794719] process done ExitSuccess
[2020-02-26 12:27:00.03875555] process done ExitSuccess
git-annex: whereis: 1 failed
```
```
git-annex version: 7.20190819+git2-g908476a9b-1~ndall+1
```

Note, that this looks the same after a failed `fsck`. I'd consider this a bug. What do you think, @datalad/developers ?

**Comment by joeyh** [comment]
This is a git-annex bug. Fix is in progress.

-- 
see shy jo

**Comment by joeyh** [comment]
Fixed for whereis.

whereis test 
  unable to query remote foo for urls: PREPARE failed with no reason given
(2 copies) 
  	7f3ad589-2933-4884-be1c-43075c8657c4 -- [foo]
   	f58d66ed-4a81-5ecf-bfa7-df66928c1154 -- [crypt]
ok

(Note that PREPARE-FAILURE replies to git-annex really should include an
error message saying what the problem is.)

-- 
see shy jo

**Comment by bpoldrack** [comment]
Thank you very much, @joeyh!

> (Note that PREPARE-FAILURE replies to git-annex really should include an error message saying what the problem is.)

Yes, I noticed that too and will fix it.

**Comment by yarikoptic** [comment]
my guess is that the fix is in [8.20200226-12-g2366e7fb8](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=2366e7fb84587150be952eba73976515238cb8dc)

**Comment by yarikoptic** [comment]
I guess it was fixed, so why to keep this one open?
